examples/spells/BoundedUInt.qnt

```diff
@@ -0,0 +1,607 @@
+
+  /// Unsafe wrapping integer addition.
+  /// Computes `lhs + rhs`, wrapping around at the boundary of the type.
+  /// Assumes lhs and rhs are valid (w.r.t. `isValid`).
+  pure def wrappingAddUnsafe(lhs: T, rhs: T): T = 
+    { 
+      v: (lhs.v + rhs.v) % (MAX + 1), 
+      error: "" 
+    }
+
+  /// Wrapping integer addition.
+  /// Computes `lhs + rhs`, wrapping around at the boundary of the type.
+  pure def wrappingAdd(lhs: T, rhs: T): T = wrapErrorBin(lhs, rhs, wrappingAddUnsafe)
+
+  /// Unsafe wrapping integer subtraction.
+  /// Computes `lhs - rhs`, wrapping around at the boundary of the type.
+  /// Assumes lhs and rhs are valid (w.r.t. `isValid`).
+  pure def wrappingSubUnsafe(lhs: T, rhs: T): T = {
+    val res = lhs.v - rhs.v
+    val adjusted = if (res < MIN) res + (MAX + 1) else res
+    { 
+      v: adjusted % (MAX + 1), 
+      error: "" 
+    }
+  }
+
+  /// Wrapping integer subtraction.
+  /// Computes `lhs - rhs`, wrapping around at the boundary of the type.
+  pure def wrappingSub(lhs: T, rhs: T): T = wrapErrorBin(lhs, rhs, wrappingSubUnsafe)
+
+  /// Unsafe wrapping integer multiplication.
+  /// Computes `lhs * rhs`, wrapping around at the boundary of the type.
+  /// Assumes lhs and rhs are valid (w.r.t. `isValid`).
+  pure def wrappingMulUnsafe(lhs: T, rhs: T): T = 
+    { 
+      v: (lhs.v * rhs.v) % (MAX + 1), 
+      error: "" 
+    }
+  
+  /// Wrapping integer multiplication.
+  /// Computes `lhs * rhs`, wrapping around at the boundary of the type.
+  pure def wrappingMul(lhs: T, rhs: T): T = wrapErrorBin(lhs, rhs, wrappingMulUnsafe)
+
+  /// Wrapping integer division.
+  /// Computes `lhs / rhs`. Wrapped division on unsigned types is just normal division. 
+  /// There’s no way wrapping could ever happen. 
+  /// This operator exists, so that all operations are accounted for in the wrapping operations.
+  pure def wrappingDiv(lhs: T, rhs: T): T = checkedDiv(lhs, rhs)
+
+  /// Wrapping integer remainder.
+  /// Computes `lhs % rhs`. Wrapped remainder on unsigned types is just normal remainder. 
+  /// There’s no way wrapping could ever happen. 
+  /// This operator exists, so that all operations are accounted for in the wrapping operations.
+  pure def wrappingRem(lhs: T, rhs: T): T = checkedRem(lhs, rhs)
+
+  /// Unsafe wrapping exponentiation.
+  /// Computes `lhs ^ rhs`, wrapping around at the boundary of the type.
+  /// Assumes lhs and rhs are valid (w.r.t. `isValid`).
+  pure def wrappingPowUnsafe(lhs: T, rhs: T): T = 
+    { 
+      v: (lhs.v ^ rhs.v) % (MAX + 1), 
+      error: "" 
+    }
+  
+  /// Wrapping exponentiation.
+  /// Computes `lhs ^ rhs`, wrapping around at the boundary of the type.
+  pure def wrappingPow(lhs: T, rhs: T): T = wrapErrorBin(lhs, rhs, wrappingPowUnsafe)
+}
+
+module BoundedUInt_Test {
+  import BoundedUInt.*
+
+  /////////////
+  // CHECKED //
+  /////////////
+
+  pure def MkCInv(x: int, y: int, op: (T,T) => T, expectedErr: str, expectedVal: (int, int) => int): bool = {
```

Comment:
- What does `MkCInv` mean? Could we add a comment or a legible name?
- `extectedVal` seems like a slightly confusing name for an operator that is not a value. Would `expectedOp` be clearer?

Reply:
- (M)a(k)e (C)hecked (Inv)ariant.
- Yeah, we wrote that name in the draft originally, then it just never got changed, I'll fix it.

Reply:
+1 for writing these out in full